src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -67,53 +68,53 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
-//      } catch (IOException ioe) {
-//        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
-//      }
-//    } else if (command instanceof AssignCommandBase) {
-//      try {
-//        storage.saveTeacherAddressBook(model.getTeacherAddressBook());
-//        storage.saveStudentAddressBook(model.getStudentAddressBook());
-//        storage.saveCourseAddressBook(model.getCourseAddressBook());
-//      } catch (IOException ioe) {
-//        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
-//      }
-//    }
-//
+    if (command instanceof AddTeacherCommand || command instanceof DeleteTeacherCommand
+        || command instanceof ClearTeacherCommand || command instanceof EditTeacherCommand) {
+      try {
+        storage.saveTeacherAddressBook(model.getTeacherAddressBook());
+      } catch (IOException ioe) {
+        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+      }
+    } else if (command instanceof AddStudentCommand || command instanceof DeleteStudentCommand
+        || command instanceof ClearStudentCommand || command instanceof EditStudentCommand) {
+      try {
+        logger.info("Attempting to save student address book");
+        storage.saveStudentAddressBook(model.getStudentAddressBook());
+      } catch (IOException ioe) {
+        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+      }
+    } else if (command instanceof AddCourseCommand || command instanceof DeleteCourseCommand
+        || command instanceof ClearCourseCommand || command instanceof EditCourseCommand) {
+      try {
+        storage.saveCourseAddressBook(model.getCourseAddressBook());
+      } catch (IOException ioe) {
+        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+      }
+    } else if (command instanceof AddFinanceCommand || command instanceof DeleteFinanceCommand
+        || command instanceof ClearFinanceCommand || command instanceof EditFinanceCommand) {
+      try {
+        storage.saveFinanceAddressBook(model.getFinanceAddressBook());
+      } catch (IOException ioe) {
+        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+      }
+    } else if (command instanceof AddAssignmentCommand || command instanceof DeleteAssignmentCommand ||
+            command instanceof ClearAssignmentCommand) {
+      try {
+        logger.info("Attempting to save assignment address book");
+        storage.saveAssignmentAddressBook(model.getAssignmentAddressBook());
+      } catch (IOException ioe) {
+        throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+      }
+    } else if (command instanceof AssignCommandBase) {
+      try {
+        storage.saveTeacherAddressBook(model.getTeacherAddressBook());
+        storage.saveStudentAddressBook(model.getStudentAddressBook());
+        storage.saveCourseAddressBook(model.getCourseAddressBook());
+        storage.saveAssignmentAddressBook(model.getAssignmentAddressBook());
```

Comment:
Why are these logics uncommented instead of using the event-driven design?

Reply:
Code is uncommented in the master, is it supposed to be like this?

Reply:
No this storage save in LogicManager is commented out because we use event-driven design.

I see that in your PR you uncommented this code part, maybe because during conflict resolve.